core/src/test/java/io/github/belgif/rest/guide/validator/core/model/MediaTypeTest.java

```diff
@@ -0,0 +1,64 @@
+package io.github.belgif.rest.guide.validator.core.model;
+
+import io.github.belgif.rest.guide.validator.core.model.helper.MediaType;
+import io.github.belgif.rest.guide.validator.core.model.helper.MediaTypeException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MediaTypeTest {
+
+    @Test
+    void testIncludes() {
+        var mediaType = new MediaType("application/json");
+        assertTrue(mediaType.includes(new MediaType("application/problem+json")));
+        assertTrue(mediaType.includes(new MediaType("application/json")));
+        assertFalse(mediaType.includes(new MediaType("application/problem+xml")));
+        assertFalse(mediaType.includes(new MediaType("*/*")));
+
+        mediaType = new MediaType("multipart/*");
+        assertTrue(mediaType.includes(new MediaType("multipart/*")));
+        assertTrue(mediaType.includes(new MediaType("multipart/form-data")));
+        assertTrue(mediaType.includes(new MediaType("multipart/chunked")));
+
+        mediaType = new MediaType("*/*");
+        assertTrue(mediaType.includes(new MediaType("*/*")));
+        assertTrue(mediaType.includes(new MediaType("application/problem+json")));
+        assertTrue(mediaType.includes(new MediaType("application/json")));
+        assertTrue(mediaType.includes(new MediaType("application/problem+xml")));
+        assertTrue(mediaType.includes(new MediaType("multipart/*")));
+        assertTrue(mediaType.includes(new MediaType("multipart/form-data")));
+        assertTrue(mediaType.includes(new MediaType("multipart/chunked")));
+        assertTrue(mediaType.includes(new MediaType("application/*")));
+    }
```

Comment:
corner case still to be tested:

"application/problem+json" doesn't include "application/json" (only the other way around)